tinygrad/shape/symbolic.py

```diff
@@ -314,10 +314,12 @@ def create_rednode(typ:Type[RedNode], nodes:List[Node]):
 @functools.lru_cache(maxsize=None)
 def sym_rename(s) -> str: return f"s{sym_rename.cache_info().currsize}"
 def sym_render(a: Union[Node, int], ops=None, ctx=None) -> str: return str(a) if isinstance(a, int) else a.render(ops, ctx)
-def sym_infer(a: Union[Node, int], var_vals: Dict[Variable, int]) -> int:
+def sym_infer(a: Union[Node, int], var_vals: Dict[Variable, int], infer_cache=None) -> int: # When infer_cache is provided, cache the infered result for **the given var_vals**.
```

Comment:
Is this cache needed, or is this just a speedup? It's not really worth trading this sort of complexity for speed

Reply:
Agree. This is an ugly hack, removed it. Still need to do something with sym_infer later, about 60-70% time of gpt2 step is spent in sym_infer for global_size and op_estimate.